modules/database/src/ioc/misc/iocshRegisterCommon.c

```diff
@@ -28,8 +28,17 @@
 /* registerAllRecordDeviceDrivers */
 static const iocshArg rrddArg0 = {"pdbbase", iocshArgPdbbase};
 static const iocshArg *rrddArgs[] = {&rrddArg0};
-static const iocshFuncDef rrddFuncDef =
-    {"registerAllRecordDeviceDrivers", 1, rrddArgs};
+static const iocshFuncDef rrddFuncDef = {
+    "registerAllRecordDeviceDrivers",
+    1,
+    rrddArgs,
+    "Register all records, devices, from all DBD available.\n\n"
+    "Calling this function is equivalent to calling every\n"
+    "'<name>_registerRecordDeviceDriver'.\n\n"
+    "These are registered into the database given as first argument,\n"
+    "which should always be 'pdbbase'.\n\n"
+    "Example: registerAllRecordDeviceDrivers pdbbase\n",
```

Comment:
Needs something about dynamic linking only.

Reply:
The essential point is that generated `*_registerRecordDeviceDriver()` code serves two purposes, one at build time and one at runtime.  `registerAllRecordDeviceDrivers()` is a alternative for the second purpose only.

When linking an executable statically, omitting generated `*_registerRecordDeviceDriver.cpp` from the build can silently result in a non-functional executable as the linker may (and likely will) discard some or all of the code being registered as unused.

I think that @anjohnson would like this to be mentioned in some way.  Maybe something like:

> Calling this function is equivalent to calling every '<name>_registerRecordDeviceDriver()' which has been linked into the process.  eg. by dynamic loading, or generated '<name>_registerRecordDeviceDriver.cpp' file(s).